mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/BlockMatrix.scala

```diff
@@ -172,6 +172,64 @@ class BlockMatrix(
     assert(cols <= nCols, s"The number of columns $cols is more than claimed $nCols.")
   }
 
+  def validate: Unit = {
+    logDebug("Validating BlockMatrix...")
+    // check if the matrix is larger than the claimed dimensions
+    try {
+      estimateDim()
+      logDebug("BlockMatrix dimensions are okay...")
+    } catch {
+      case exc: AssertionError => throw new SparkException(s"$exc\nPlease instantiate a " +
+        s"new BlockMatrix with the correct dimensions.")
+      case e: Exception =>
+        throw new SparkException(s"${e.getMessage}\n${e.getStackTraceString}")
+    }
+    // Check if there are multiple MatrixBlocks with the same index.
+    val indexCounts = blocks.countByKey().filter(p => p._2 > 1)
+    if (indexCounts.size > 50) {
```

Comment:
I don't understand this part. It is hard to guess what went wrong if there are more than 50 duplicates. I would rather leave it to users to find out. The code in my previous comment should be sufficient.